--- listdir.py.old	2013-04-01 12:28:37.000000000 +0200
+++ listdir.py	2013-04-01 14:51:29.000000000 +0200
@@ -2,6 +2,7 @@
 from __future__ import unicode_literals, division, absolute_import
 import os
 import logging
+import sys
 from flexget.plugin import register_plugin
 from flexget.entry import Entry
 
@@ -33,6 +34,8 @@
         for path in config:
             path = os.path.expanduser(path)
             for name in os.listdir(unicode(path)):
+                # Fix non latin-1 encoding filenames
+                name = name.encode(sys.getfilesystemencoding()).decode('latin1')
                 e = Entry()
                 e['title'] = name
                 filepath = os.path.join(path, name)
